test/modules/Modules.Test/TestResultCoordinator/Reports/DirectMethod/DirectMethodLongHaulReportGeneratorTest.cs

```diff
@@ -0,0 +1,394 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Modules.Test.TestResultCoordinator.Reports.DirectMethod
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Net;
+    using System.Threading.Tasks;
+    using global::TestResultCoordinator.Reports;
+    using global::TestResultCoordinator.Reports.DirectMethod.LongHaul;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil.TestResults;
+    using Microsoft.Azure.Devices.Edge.Storage;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Moq;
+    using Newtonsoft.Json;
+    using Xunit;
+
+    public class DirectMethodLongHaulReportGeneratorTest
+    {
+        static readonly string TestDescription = "dummy description";
+
+        [Fact]
+        public void TestConstructorSuccess()
+        {
+            string senderSource = "senderSource";
+            Option<string> receiverSource = Option.Some("receiverSource");
+            int batchSize = 10;
+            string resultType = "resultType1";
+
+            var mockSenderStore = new Mock<ISequentialStore<TestOperationResult>>();
+            var senderResults = new StoreTestResultCollection<TestOperationResult>(mockSenderStore.Object, batchSize);
+            var mockReceiverStore = new Mock<ISequentialStore<TestOperationResult>>();
+            var receiverResults = Option.Some<ITestResultCollection<TestOperationResult>>(
+                new StoreTestResultCollection<TestOperationResult>(mockReceiverStore.Object, batchSize));
+
+            var reportGenerator = new DirectMethodLongHaulReportGenerator(
+                TestDescription,
+                Guid.NewGuid().ToString(),
```

Comment:
will it throw if this guid isn't provided?

Reply:
yes. that test exists too - TestConstructorThrowsWhenTrackingIdIsNotProvided